server/sonar-process-monitor/src/main/java/org/sonar/application/cluster/AppStateClusterImpl.java

```diff
@@ -160,21 +180,66 @@ public void reset() {
   @Override
   public void close() {
     if (hzInstance != null) {
-      operationalProcesses.removeEntryListener(listenerUuid);
+      // Removing listeners
+      operationalProcesses.removeEntryListener(operationalProcessListenerUUID);
+
+      // Removing the operationalProcess from the replicated map
       operationalProcesses.keySet().forEach(
         clusterNodeProcess -> {
           if (clusterNodeProcess.getNodeUuid().equals(getLocalUuid())) {
             operationalProcesses.remove(clusterNodeProcess);
           }
         });
+
+      // Shutdown Hazelcast properly
       hzInstance.shutdown();
     }
   }
 
-  private String getLocalUuid() {
+  String getLocalUuid() {
     return hzInstance.getLocalEndpoint().getUuid();
   }
 
+  String getHostname() {
+    String hostname;
+    String ips;
+    try {
+      hostname = InetAddress.getLocalHost().getHostName();
+    } catch (UnknownHostException e) {
+      hostname = "unresolved hostname";
+    }
+
+    try {
+      ips = list(NetworkInterface.getNetworkInterfaces()).stream()
+        .flatMap(netif ->
+          list(netif.getInetAddresses()).stream()
+            .filter(inetAddress ->
+                // Removing IPv6 for the time being
+                inetAddress instanceof Inet4Address &&
+                // Removing loopback addresses, useless for identifying a server
+                !inetAddress.isLoopbackAddress() &&
+                // Removing interfaces without IPs
+                !isBlank(inetAddress.getHostAddress())
+            )
+            .map(InetAddress::getHostAddress)
+        )
+        .filter(p -> !isBlank(p))
+        .collect(Collectors.joining(","));
+    } catch (SocketException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)